src/modules/dpos/validator/components/RegisterDelegateForm/index.js

```diff
@@ -1,82 +1,56 @@
-import React, { useState, useRef, useEffect } from 'react';
-import { useSelector } from 'react-redux';
+import React from 'react';
 import { useTranslation } from 'react-i18next';
-import { regex } from 'src/const/regex';
 import { MODULE_ASSETS_NAME_ID_MAP } from '@transaction/configuration/moduleAssets';
-import { getDelegate } from '@dpos/validator/api';
 import BoxHeader from 'src/theme/box/header';
 import BoxContent from 'src/theme/box/content';
 import { Input } from 'src/theme';
-import Tooltip from 'src/theme/Tooltip';
 import TxComposer from '@transaction/components/TxComposer';
+import useDelegateName from '../../hooks/useDelegateName';
+import useDelegateKey from '../../hooks/useDelegateKey';
+import InputLabel from './InputLabel';
 import styles from './form.css';
 
-const validateUsername = (query, t) => {
-  if (query.length === 0) {
-    return t('Username can not be empty.');
-  }
-  if (query.length < 2) {
-    return t('Username is too short.');
-  }
-  if (query.length > 20) {
-    return t('Username is too long.');
-  }
-  const hasInvalidChars = query.replace(regex.delegateSpecialChars, '');
-  if (hasInvalidChars) {
-    return t(`Invalid character ${hasInvalidChars.trim()}`);
-  }
-  return '';
-};
+const isFormValid = (name, generatorPublicKey, blsPublicKey, proofOfPossession) => (
+  !name.error && !name.loading && !generatorPublicKey.error
+  && !blsPublicKey.error && !proofOfPossession.error
+);
 
-// eslint-disable-next-line max-statements
 const RegisterDelegateForm = ({
   nextStep,
   prevState,
 }) => {
-  const timeout = useRef();
   const { t } = useTranslation();
-  const network = useSelector(state => state.network);
-  const [error, setError] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [username, setUsername] = useState(prevState?.rawTx?.asset.username ?? '');
+  const [name, setName] = useDelegateName(prevState?.rawTx?.asset.username);
+  const [generatorPublicKey, blsPublicKey, proofOfPossession, setKey] = useDelegateKey(
```

Comment:
recently we introduce [react-hooks-form](https://react-hook-form.com) please use that one so it dosn't need to create custom validation you can use [yup validation](https://github.com/jquense/yup#stringmatchesregex-regex-message-string--function-schema)